### PR TITLE
header: verify aggregator hash against validators

### LINKED ISSUE #833: Add Verification of the validator set in the Full node to check against the validator set in the header.
We need to check if the validator set matches the one in the header. If not, the header is fraudulent, and a header fraud-proof must be triggered. This can panic for now, but we need to understand what the intended behaviour is.

### PR DESCRIPTION
**Pull request by tuxcanfly**
Fixes #833

### CODE REVIEW

**Review by nashqueue — APPROVED** [review]
Great!

File: node/full_client_test.go [hunk 1]
```diff
@@ -56,14 +56,15 @@ func getRandomValidatorSet() *tmtypes.ValidatorSet {
 	}
 }
 
+var genesisValidatorKey = ed25519.GenPrivKey()
+
 // TODO: use n and return n validators
 func getGenesisValidatorSetWithSigner(n int) ([]tmtypes.GenesisValidator, crypto.PrivKey) {
-	validatorKey := ed25519.GenPrivKey()
```
Comment (gupadhyaya) [inline node/full_client_test.go:61]: why does this not work and simply using the global variable works?

Comment (tuxcanfly) [reply node/full_client_test.go:61]: This was creating a new validator key everytime this function was being called, so the nodes had different validator sets. The aggregator hash was being set from the first node (in this test case first node is aggregator) and the rest of the nodes were failing the hash check.

Comment (gupadhyaya) [reply node/full_client_test.go:61]: got it. may be you can remove the function `getGenesisValidatorSetWithSigner` and have GEN_VAL_SET and GEN_KEY as constants?

Comment (tuxcanfly) [reply node/full_client_test.go:61]: Created #860

**Review by gupadhyaya — APPROVED** [review]

**Pull request merged**